command/v6/service_access_command_test.go

```diff
@@ -173,30 +264,52 @@ var _ = Describe("service-access Command", func() {
+				When("there are multiple brokers", func() {
+					var brokers []v2action.ServiceBrokerSummary
 
-				BeforeEach(func() {
-					brokers = generateBrokers(2)
-					fakeActor.GetServiceBrokerSummariesReturns(generateBrokers(2), v2action.Warnings{"warning"}, nil)
+					BeforeEach(func() {
+						brokers = generateBrokers(2)
+						fakeActor.GetServiceBrokerSummariesReturns(generateBrokers(2), v2action.Warnings{"warning"}, nil)
+					})
+
+					It("displays each service broker, service, plan and access with org", func() {
+						Expect(executeErr).ToNot(HaveOccurred())
+
+						tableHeaders := `service\s+plan\s+access\s+orgs`
+						Expect(testUI.Out).To(Say(`broker:\s+%s`, brokers[0].Name))
+						Expect(testUI.Out).To(Say(tableHeaders))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 1, 1, "all")))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 1, 0, "none")))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 0, 1, "all")))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 0, 0, "limited")))
+						Expect(testUI.Out).To(Say(`broker:\s+%s`, brokers[1].Name))
+						Expect(testUI.Out).To(Say(tableHeaders))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 1, 1, "all")))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 1, 0, "none")))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 0, 1, "all")))
+						Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 0, 0, "limited")))
+					})
 				})
 
-				It("displays each service broker, service, plan and access with org in the correct position", func() {
-					Expect(executeErr).ToNot(HaveOccurred())
-
-					tableHeaders := `service\s+plan\s+access\s+orgs`
-					Expect(testUI.Out).To(Say(`broker:\s+%s`, brokers[0].Name))
-					Expect(testUI.Out).To(Say(tableHeaders))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 1, 1, "all")))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 1, 0, "none")))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 0, 1, "all")))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 0, 0, 0, "limited")))
-					Expect(testUI.Out).To(Say(`broker:\s+%s`, brokers[1].Name))
-					Expect(testUI.Out).To(Say(tableHeaders))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 1, 1, "all")))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 1, 0, "none")))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 0, 1, "all")))
-					Expect(testUI.Out).To(Say(rowMatcher(brokers, 1, 0, 0, "limited")))
+				When("the summaries returned are unordered", func() {
+					BeforeEach(func() {
+						unorderedBrokers := []v2action.ServiceBrokerSummary{createBroker(2), createBroker(1)}
+						fakeActor.GetServiceBrokerSummariesReturns(unorderedBrokers, v2action.Warnings{"warning"}, nil)
+					})
+
+					It("sorts brokers and services before displaying", func() {
+						Expect(executeErr).ToNot(HaveOccurred())
+
+						// Unfortuantely we cannot use the brokers struct here as the command does an in place sort
```

Comment:
I'm unclear as to how the ordering in this case would end up being different than in the test above. Can you clarify?

Reply:
The ordering would be different because we were passing in the brokers rather than generating a duplicate of the slice, and then it gets sorted in place. However, we realised that the only extra thing this was testing was the sorting of the brokers, because the `generateBroker` function was already producing an out of order structure, so we just deleted this test and in the one above also checked that sorting of brokers works.